docs/user-guide/kubectl-conventions.md

```diff
@@ -8,15 +8,45 @@ assignees:
 * TOC
 {:toc}
 
-## Using `kubectl` in Reusable Scripts 
+## Using `kubectl` in Reusable Scripts
 
 If you need stable output in a script, you should:
 
-* Request one of the machine-oriented output forms, such as `-o name`, `-o json`, `-o yaml`, `-o go-template`, or `-o jsonpath` 
+* Request one of the machine-oriented output forms, such as `-o name`, `-o json`, `-o yaml`, `-o go-template`, or `-o jsonpath`
 * Specify `--output-version`, since those output forms (other than `-o name`) output the resource using a particular API version
 * Specify `--generator` to pin to a specific behavior forever, if using generator-based commands (such as `kubectl run` or `kubectl expose`)
 * Don't rely on context, preferences, or other implicit state
 
+## Generators
+
+`kubectl run` provides following generators:
+
+* `run/v1` - creates a replication controller.
+* `run-pod/v1` - creates a pod.
+* `deployment/v1beta1` - creates a deployment.
+* `job/v1beta1` - creates a job (using `extensions/v1beta1' endpoint).
```

Comment:
Perhaps we should put the object first, generator later, something like:

Use `--generator` flag to specify the object to be generated, for example, to generate a:
* Replication Controller: `--generator=run/v1` (or just `run/v1` is enough?)
* Pod: `--generator=run-pod/v1` 
...

Or maybe use a table?